src/UserSpaceInstrumentation/AllocateInTraceeTest.cpp

```diff
@@ -65,14 +66,25 @@ TEST(AllocateInTraceeTest, AllocateAndFree) {
 
   // Allocate a megabyte in the tracee.
   constexpr uint64_t kMemorySize = 1024 * 1024;
-  auto result_allocate = AllocateInTracee(pid, kMemorySize);
-  CHECK(result_allocate.has_value());
-
+  auto result_allocate = AllocateInTracee(pid, 0, kMemorySize);
+  ASSERT_TRUE(result_allocate.has_value());
```

Comment:
add a test verifying that AllocateInTracee works as expected when allocated twice at the same address?

Reply:
While looking into this I recognized that it makes sense to move all the stuff that handeles AddressRange's into a separate header. There is already another definition of `AddressRange` in AccessTraceesMemory.h. I need to get rid of that. This will also make testing this easier.
I don't want to squeeze these changes into this PR. So let me do this in a followup.